src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -6,19 +6,180 @@
+    private static final int LOAD_FACTOR = 75;
+    private Node<K, V>[] buckets;
+    private int size;
+
+    public MyHashMap() {
+        size = 0;
+        buckets = new Node[DEFAULT_ARRAY_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (ensureCapacity(size)) {
+            resize();
+        }
+        if (key == null) {
+            putForNullKey(value);
+            return;
+        }
+        int index = index(key);
+        if (buckets[index] == null) {
+            buckets[index] = new Node(key, value);
+            size++;
+            return;
+        }
+        if (buckets[index].value == null) {
+            buckets[index] = new Node(key, value);
+            size++;
+        } else {
+            setBucket(index, key, value);
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        if (key == null) {
+            return getForNullKey();
+        }
+        int index = index(key);
+        return getBucket(index, key);
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    //////////////////////My_Methods///////////////////////////////////
+
+    private int index(K key) {
+        return Math.abs(key.hashCode() % (buckets.length));
+    }
+
+    private boolean ensureCapacity(int size) {
+        return size > (buckets.length * LOAD_FACTOR) / 100;
+    }
+
+    private void putForNullKey(V value) {
+        if (buckets[0] == null) {
+            buckets[0] = new Node(null, value);
+            size++;
+            return;
+        }
+        if (buckets[0].key == null) {
+            buckets[0].value = value;
+            return;
+        }
+        Node<K, V> currentNode = getNoge(buckets[0]);
+        while (currentNode.next != null) {
+            if (currentNode.next.key == null) {
+                currentNode.next.value = value;
+                return;
+            }
+            currentNode = currentNode.next;
+        }
+        Node<K, V> entry = new Node(value);
+        currentNode.next = entry;
+        size++;
+    }
+
+    private V getForNullKey() {
+        if (buckets[0] == null) {
+            return null;
+        }
+        if (buckets[0].key == null) {
+            return buckets[0].value;
+        }
+        Node<K, V> currentNode = buckets[0];
+        while (currentNode.next != null) {
+            if (currentNode.next.key == null) {
+                return currentNode.next.value;
+            }
+            currentNode = currentNode.next;
+        }
+        return null;
+    }
+
+    private void resize() {
+        Node<K, V>[] oldData = buckets;
+        buckets = new Node[buckets.length * 2];
+        size = 0;
+        for (int i = 0; i < oldData.length; i++) {
+            if (oldData[i] != null) {
+                putLL(oldData[i]);
+            }
+        }
+    }
+
+    private void setValue(int index, K key, V value) {
```

Comment:
it seems that you use this method only once in program, in this case it's not worth to make separate method